### PR TITLE
GLS matrix and rhs assembler for non Newtonian flows

### PR DESCRIPTION
**Pull request by CaroleAnneDaunais**
A non Newtonian assembler has previously been added to the Grad-Div Navier-Stokes solver. It is now added to the GLS Navier-Stokes solver, the same way it was done in the DG solver, with an extra term in the strong residual to encounter for the viscosity gradient (which is not zero for a non Newtonian flow). 
Two tests were added to recreate the gls_navier_stokes_2d/mms2d-unstructured_gls test, this time with a structured mesh and a viscosity follower the Carreau rheological model. One of them has first order lagragian polynomial for velocity, the other one is second order in velocity. Both are first order in pressure.

### CODE REVIEW

File: include/solvers/post_processors.h [hunk 1]
```diff
@@ -193,18 +193,21 @@ class NonNewtonianViscosityPostprocessor : public DataPostprocessorScalar<dim>
   evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &inputs,
                         std::vector<Vector<double>> &computed_quantities) const
   {
-    Tensor<2, dim>     shear_rate;
+    double             shear_rate_magnitude;
```
Comment (blaisb) [inline include/solvers/post_processors.h:196]:
```suggestion
```

File: include/solvers/post_processors.h [hunk 2]
```diff
@@ -193,18 +193,21 @@ class NonNewtonianViscosityPostprocessor : public DataPostprocessorScalar<dim>
...
     const unsigned int n_quadrature_points = inputs.solution_gradients.size();
 
     for (unsigned int q = 0; q < n_quadrature_points; ++q)
       {
-        shear_rate = calculate_shear_rate(inputs.solution_gradients[q]);
-        computed_quantities[q] = rheological_model->get_viscosity(shear_rate);
+        shear_rate_magnitude =
```
Comment (blaisb) [inline include/solvers/post_processors.h:201]:
```suggestion
        const double shear_rate_magnitude =
```

**Review by blaisb — COMMENTED** [review]
Very minor stuff.

File: include/solvers/navier_stokes_scratch_data.h [hunk 3]
```diff
@@ -226,6 +226,8 @@ class NavierStokesScratchData
       current_solution, this->velocity_gradients);
     this->fe_values[velocities].get_function_laplacians(
       current_solution, this->velocity_laplacians);
+    this->fe_values[velocities].get_function_hessians(current_solution,
```
[the remark this answers is not in the document]
Comment (CaroleAnneDaunais) [reply include/solvers/navier_stokes_scratch_data.h]: Fixed

**Review by blaisb — APPROVED** [review]
Good to merge once CI passes.

**Pull request merged**